freshclam/freshclam.c

```diff
@@ -1475,7 +1475,38 @@ fc_error_t perform_database_update(
          * Prune database directory of official databases
          * that are no longer available or no longer desired.
          */
-        (void)fc_prune_database_directory(databaseList, nDatabases);
+        uint32_t i;
+
+        // include the URL databases in the prune process
+        char **newDatabaseList = (char **)malloc(sizeof(char *) * (nDatabases + nUrlDatabases));
+        if (NULL == newDatabaseList) {
+            logg(LOGG_ERROR, "perform_database_update: Can't allocate memory for newDatabaseList\n");
+            status = FC_EMEM;
+            goto done;
+        }
+
+        for (i = 0; i < nDatabases; i++) {
+            newDatabaseList[i] = strdup(databaseList[i]);
+        }
+        uint32_t nNewDatabases = nDatabases;
+
+        for (i = 0; i < nUrlDatabases; i++) {
+            // only append the URL databases that end with '.cvd'
+            if (strlen(urlDatabaseList[i]) > 4 && 0 == strcasecmp(urlDatabaseList[i] + strlen(urlDatabaseList[i]) - 4, ".cvd")) {
+                const char* startOfFilename = strrchr(urlDatabaseList[i], '/') + 1;
```

Comment:
Where do the urls come from?  Is it possible that they would be malformed and not contain a '/'?

Reply:
The URL's come from the freshclam.conf `DatabaseCustomURL` option. It would be a misconfiguration. Could happen.